services/src/main/java/org/keycloak/crypto/CryptoUtils.java

```diff
@@ -20,33 +20,59 @@
 import org.keycloak.models.KeycloakSession;
 import org.keycloak.provider.ProviderFactory;
 
+import java.security.Key;
+import java.security.PrivateKey;
+import java.security.PublicKey;
 import java.util.AbstractMap;
 import java.util.List;
 import java.util.Map;
 import java.util.stream.Collectors;
 
 /**
  * Utility class for common cryptographic operations and algorithm discovery.
- * 
+ *
  * @author <a href="https://github.com/forkimenjeckayang">Forkim Akwichek</a>
  */
 public class CryptoUtils {
 
-    /**
-     * Returns the supported asymmetric signature algorithms.
-     * This method discovers all available SignatureProvider implementations and filters
-     * for those that support asymmetric algorithms (RSA, EC, EdDSA, etc.).
-     * 
-     * @param session The Keycloak session
-     * @return List of asymmetric signature algorithm names
-     */
-    public static List<String> getSupportedAsymmetricSignatureAlgorithms(KeycloakSession session) {
-        return session.getKeycloakSessionFactory().getProviderFactoriesStream(SignatureProvider.class)
-                .map(ProviderFactory::getId)
-                .map(algorithm -> new AbstractMap.SimpleEntry<>(algorithm, session.getProvider(SignatureProvider.class, algorithm)))
-                .filter(entry -> entry.getValue() != null)
-                .filter(entry -> entry.getValue().isAsymmetricAlgorithm())
-                .map(Map.Entry::getKey)
-                .collect(Collectors.toList());
-    }
+	/**
+	 * Returns the supported asymmetric signature algorithms.
+	 * This method discovers all available SignatureProvider implementations and filters
+	 * for those that support asymmetric algorithms (RSA, EC, EdDSA, etc.).
+	 *
+	 * @param session The Keycloak session
+	 * @return List of asymmetric signature algorithm names
+	 */
+	public static List<String> getSupportedAsymmetricSignatureAlgorithms(KeycloakSession session) {
```

Comment:
It seems that there is no real change in this method in your PR, but it is shown as changed. Can you please remove this change from your PR?

I think the problem is, that you updated this to use tabs instead of 4 spaces. Please note that it is recommended in the Keycloak codebase to use spaces instead of tabs and use the align of 4 spaces. I suggest you update your IDE to use 4 spaces instead of tabs. This should remove the changes of the method `getSupportedAsymmetricSignatureAlgorithms` from your PR and keep just new method `getSupportedAsymmetricEncryptionAlgorithms`

Reply:
done